packages/dashboard/src/app.tsx

```diff
@@ -2,187 +2,101 @@ import '@fontsource/roboto/300.css';
-        </Routes>
-      )}
-    </ResourcesContext.Provider>
-  ) : null;
+import { appConfig } from './app-config';
+import { InitialWindow, LocallyPersistentWorkspace, RmfDashboard, Workspace } from './components';
+import { MicroAppManifest } from './components/micro-app';
+import doorsApp from './micro-apps/doors-app';
+import liftsApp from './micro-apps/lifts-app';
+import createMapApp from './micro-apps/map-app';
+import robotMutexGroupsApp from './micro-apps/robot-mutex-groups-app';
+import robotsApp from './micro-apps/robots-app';
+import tasksApp from './micro-apps/tasks-app';
+import StubAuthenticator from './services/stub-authenticator';
+
+const mapApp = createMapApp({
+  attributionPrefix: appConfig.attributionPrefix,
+  defaultMapLevel: appConfig.defaultMapLevel,
+  defaultRobotZoom: appConfig.defaultRobotZoom,
+  defaultZoom: appConfig.defaultZoom,
+});
+
+const appRegistry: MicroAppManifest[] = [
+  mapApp,
+  doorsApp,
+  liftsApp,
+  robotsApp,
+  robotMutexGroupsApp,
+  tasksApp,
+];
+
+const homeWorkspace: InitialWindow[] = [
+  {
+    layout: { x: 0, y: 0, w: 12, h: 6 },
+    microApp: mapApp,
+  },
+];
+
+const robotsWorkspace: InitialWindow[] = [
+  {
+    layout: { x: 0, y: 0, w: 7, h: 4 },
+    microApp: robotsApp,
+  },
+  { layout: { x: 8, y: 0, w: 5, h: 8 }, microApp: mapApp },
+  { layout: { x: 0, y: 0, w: 7, h: 4 }, microApp: doorsApp },
+  { layout: { x: 0, y: 0, w: 7, h: 4 }, microApp: liftsApp },
+  { layout: { x: 8, y: 0, w: 5, h: 4 }, microApp: robotMutexGroupsApp },
+];
+
+const tasksWorkspace: InitialWindow[] = [
+  { layout: { x: 0, y: 0, w: 7, h: 8 }, microApp: tasksApp },
+  { layout: { x: 8, y: 0, w: 5, h: 8 }, microApp: mapApp },
+];
+
+export default function App() {
+  return (
+    <RmfDashboard
+      apiServerUrl="http://localhost:8000"
+      trajectoryServerUrl="http://localhost:8006"
+      authenticator={new StubAuthenticator()}
+      helpLink={appConfig.helpLink}
+      reportIssueLink={appConfig.reportIssue}
+      resources={appConfig.resources.default}
+      tasks={{
+        allowedTasks: appConfig.allowedTasks,
+        pickupZones: appConfig.pickupZones,
+        cartIds: appConfig.cartIds,
+      }}
+      tabs={[
+        {
+          name: 'Map',
+          route: '',
+          element: <Workspace initialWindows={homeWorkspace} />,
+        },
+        {
+          name: 'Robots',
+          route: 'robots',
+          element: <Workspace initialWindows={robotsWorkspace} />,
+        },
+        {
+          name: 'Tasks',
+          route: 'tasks',
+          element: <Workspace initialWindows={tasksWorkspace} />,
+        },
+        {
+          name: 'Custom',
+          route: 'custom',
+          element: (
+            <LocallyPersistentWorkspace
+              defaultWindows={[]}
+              allowDesignMode
+              appRegistry={appRegistry}
+              storageKey="custom-workspace"
+            />
+          ),
+        },
```

Comment:
I think we'll need to find a way to toggle the custom tab based on `APP_CONFIG_ENABLE_CUSTOM_TABS`, as well as the admin tab with `APP_CONFIG_ENABLE_ADMIN_TAB`

Or was the whole idea to remove the config variables, and let users set up the dashboard however they want with this new way? If so, then the admin tab should also be added in

or is the admin tab left out because we haven't worked out authz and actions mapping properly?

Reply:
Yeah the plan is to have no build time config variables, after this transitions to a framework library, we can't have build time variables unless we integrate the build system into the framework as well.

Whether to have custom tabs can be easily configured by downstream with `allowDesignMode`, they can also configure the number of custom tabs and if "builtin" tabs can be customized.

The admin tab is more tricky, until the backend supports proper authz, we should never show the admin tab. But when it does support authz, we should always show the admin tab if the user is an admin. The problem is that in terms of DX, it may be disruptive to force include a tab that the user did not configure. Most likely it will not be a "tab" and instead be accessed from the user icon similar to the admin panel.